lib/Drush/Boot/DrupalBoot8.php

```diff
@@ -64,44 +73,13 @@ function bootstrap_drupal_full() {
   }
 
   /**
-   * Ensures Drupal 8 is bootstrapped to the specified phase.
-   *
-   * In order to bootstrap Drupal from another PHP script, you can use this code:
-   * @code
-   *   require_once '/path/to/drupal/core/vendor/autoload.php';
-   *   require_once '/path/to/drupal/core/includes/bootstrap.inc';
-   *   drupal_bootstrap(DRUPAL_BOOTSTRAP_FULL);
-   * @endcode
-   *
-   * @param $phase
-   *   A constant telling which phase to bootstrap to. Possible values:
-   *   - DRUSH_BOOTSTRAP_DRUPAL_CONFIGURATION: Initializes configuration and
-   *     kernel.
-   *   - DRUSH_BOOTSTRAP_DRUPAL_FULL: Boots the kernel.
+   * {@inheritdoc}
    */
-  function drupal8_bootstrap($phase = NULL) {
-    // Temporary variables used for booting later legacy phases.
-    /** @var \Drupal\Core\DrupalKernel $kernel */
-    static $kernel;
-    static $boot_level = 0;
-
-    if (isset($phase)) {
-      $request = Request::createFromGlobals();
-      for ($current_phase = $boot_level; $current_phase <= $phase; $current_phase++) {
-
-        switch ($current_phase) {
-          case DRUSH_BOOTSTRAP_DRUPAL_CONFIGURATION:
-            $classloader = drush_drupal_load_autoloader(DRUPAL_ROOT);
-            $kernel = DrupalKernel::createFromRequest($request, $classloader, 'prod');
-            break;
-
-          case DRUSH_BOOTSTRAP_DRUPAL_FULL:
-            $kernel->boot();
-            $kernel->prepareLegacyRequest($request);
-            break;
-        }
-      }
-      $boot_level = $phase;
+  public function terminate() {
+    parent::terminate();
+
+    if ($this->kernel) {
```

Comment:
$this->kernel is not initialized until DRUSH_BOOTSTRAP_DRUPAL_CONFIGURATION, but it is possible that a DrupalBoot8 might be constructed and terminated without ever reaching this phase, so I think the if is necessary.